src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.cs

```diff
@@ -150,6 +150,8 @@ public sealed override byte[] GetSubjectKeyIdentifier(X509Certificate2 certifica
                                 return (T)(object)new ECDsaCng(cngKey);
                             if (typeof(T) == typeof(DSA))
                                 return (T)(object)new DSACng(cngKey);
+                            if (typeof(T) == typeof(MLDsa))
+                                throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_AlgorithmNotSupported, nameof(MLDsa)));
```

Comment:
Why isn't this `return (T)(object)new MLDsaCng(cngKey)`?

The SLH-DSA one is "because it predated the type" (and so it looks like it could use an update), but this one looks accomplishable.

Not that I can tell at a glance what path this gets used in.